vcd/resource_vcd_org_vdc_access_control.go

```diff
@@ -0,0 +1,236 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"strings"
+)
+
+func resourceVcdOrgVdcAccessControl() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVdcVdcAccessControlCreateUpdate,
+		ReadContext:   resourceVdcVdcAccessControlRead,
+		UpdateContext: resourceVdcVdcAccessControlCreateUpdate,
+		DeleteContext: resourceVdcVdcAccessControlDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVdcVdcAccessControlImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"shared_with_everyone": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Whether the VDC is shared with everyone",
+			},
+			"everyone_access_level": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{types.ControlAccessReadOnly}, true),
+				Description:  "Access level when the VDC is shared with everyone (only ReadOnly is available). Required when shared_with_everyone is set",
+			},
+			"shared_with": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"user_id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "ID of the user to which we are sharing. Required if group_id is not set",
```

Comment:
here terraform validation can be added `oneOf` or different which doesn't allow both userid and groupid

Reply:
Not sure how to do that since it is a set. I've tried but Terraform doesn't like it. I've check other resources like `vApp access control`and doesn't have it.

Reply:
Seems we can't have additional validation in Set.